lumigator/jobs/inference/inference.py

```diff
@@ -114,8 +114,8 @@ def run_inference(config: InferenceJobConfig, api_key: str | None = None) -> Pat
 
     prediction_results, inference_time = predict(dataloader_iterable, model_client)
     output[config.job.output_field] = [p.prediction for p in prediction_results]
-    output["reasoning"] = [p.reasoning for p in prediction_results]
-    output["inference_metrics"] = [p.metrics for p in prediction_results]
+    output["reasoning"] = [p.reasoning for p in prediction_results if p.reasoning is not None]
```

Comment:
Could this cause mismatch of lists for different predictions?

Reply:
Good point @javiermtorres, thanks. I'll catch up with @njbrake and get his thoughts but I think you might be right. 

I've removed the change for now though 😄